scipy/stats/tests/test_qmc.py

```diff
@@ -445,63 +457,50 @@ class TestHalton(QMCEngineTests):
 class TestLHS(QMCEngineTests):
     qmce = qmc.LatinHypercube
     can_scramble = False
-    unscramble_nd = np.array([[0.73412877, 0.50416027],
-                              [0.5924405, 0.51284543],
-                              [0.57790629, 0.70797228],
-                              [0.44357794, 0.64496811],
-                              [0.23461223, 0.55712172],
-                              [0.45337347, 0.4440004],
-                              [0.73381992, 0.01751516],
-                              [0.52245145, 0.33099331]])
+
+    scramble = [True, False]
+    ids = ["Scrambled", "Unscrambled"]
 
     def test_continuing(self, *args):
         pytest.skip("Not applicable: not a sequence.")
 
     def test_fast_forward(self, *args):
         pytest.skip("Not applicable: not a sequence.")
 
-    def test_sample_centered(self):
-        engine = self.engine(d=2, scramble=False, centered=True)
-        sample = engine.random(n=5)
-        out = np.array([[0.3, 0.5],
-                        [0.5, 0.3],
-                        [0.1, 0.7],
-                        [0.7, 0.7],
-                        [0.7, 0.1]])
-        assert_almost_equal(sample, out, decimal=1)
+    def test_sample(self, *args):
+        pytest.skip("Not applicable: the value of reference sample is implementation dependent.")
 
+    def test_continuing(self, *args):
+        pytest.skip("Not applicable: the value of reference sample is implementation dependent.")
 
-class TestOLHS(QMCEngineTests):
-    qmce = qmc.OrthogonalLatinHypercube
-    can_scramble = False
-    unscramble_nd = np.array([[0.01587123, 0.01618008],
-                              [0.24583973, 0.35254855],
-                              [0.66702772, 0.82434795],
-                              [0.80642206, 0.89219419],
-                              [0.2825595, 0.41900669],
-                              [0.98003189, 0.52861091],
-                              [0.54709371, 0.23248484],
-                              [0.48715457, 0.72209797]])
+    def test_sample_stratified(self):
+        d, n = 4, 20
+        expected1d = (np.arange(n) + 0.5) / n
+        expected = np.broadcast_to(expected1d, (d, n)).T
 
-    def test_continuing(self, *args):
-        pytest.skip("Not applicable: not a sequence.")
+        engine = self.engine(d=d, scramble=False, centered=True)
+        sample = engine.random(n=n)
+        sorted_sample = np.sort(sample, axis=0)
 
-    def test_fast_forward(self, *args):
-        pytest.skip("Not applicable: not a sequence.")
+        assert_almost_equal(sorted_sample, expected)
+        assert np.any(sample != expected)
+
+        engine = self.engine(d=d, scramble=False, centered=False)
+        sample = engine.random(n=n)
+        sorted_sample = np.sort(sample, axis=0)
+        assert_allclose(sorted_sample, expected, atol=0.5/n)
+        assert np.any(sample - expected > 0.5/n)
+
+    @pytest.mark.parametrize("scramble", scramble, ids=ids)
```

Comment:
```suggestion
    @pytest.mark.parametrize("scramble", self.scramble, ids=self.ids)
```

Reply:
I get the following error...
```
NameError: name 'self' is not defined
```

Reply:
mmm maybe `cls` then. Sorry did not test it locally.

Reply:
`cls` does not work as well... I am not sure if we can access the information of the class outside the methods.